File: Swift/two-sum.swift
/**
 * Problem Link: https://leetcode.com/problems/two-sum/
 */

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict = [Int: Int]()
        for n in 0..<nums.count {
            if dict[nums[n]] != nil {
                return [dict[nums[n]]!, n]
            } else {
                dict[target - nums[n]] = n
            }
        }
        return [Int]()
    }
}

// Wed Mar 25 14:03:04 PDT 2020
class Solution {
    /// Go through every element in the array.
    /// While remembering the index of every element visited.
    /// If target - nums[index] is in the indexMap, which means that element has been visited before.
    /// then, return the indices of these two.
    /// otherwise, keep visiting every element, while remember it and its index.
    ///
    /// - Complexity: O(n), n is the number of element in the array.
    ///
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var indexMap: [Int : Int] = [:]
        for index in 0 ..< nums.count {
            if let pIndex = indexMap[target - nums[index]] {
                return [pIndex, index]
            }
            indexMap[nums[index]] = index
        }
        return [0, 0]
    }
}

File: Python/two-sum.py
/**
 * https://leetcode.com/problems/two-sum/
 * 
 * 
 */ 

// Date Sun Mar 23 16:10:39 PDT 2025
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for start in range(len(nums)):
            for end in range(start + 1, len(nums)):
                if nums[start] + nums[end] == target:
                    return [start, end]
        return []
/**
 * https://leetcode.com/problems/two-sum/
 * 
 * 
 */ 
// Date: Thu Mar 27 16:50:15 PDT 2025
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for index in range(len(nums)):
            left = target - nums[index]
            if left in hashmap:
                return [hashmap[left], index]
            hashmap[nums[index]] = index
        return []

